modules/core/core-publisher/src/main/java/org/eclipse/dirigible/core/publisher/api/PublisherHandler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.dirigible.core.publisher.api;
+
+import org.eclipse.dirigible.commons.config.StaticObjects;
+import org.eclipse.dirigible.core.scheduler.api.SchedulerException;
+import org.eclipse.dirigible.database.persistence.PersistenceManager;
+import org.eclipse.dirigible.database.sql.SqlFactory;
+import org.eclipse.dirigible.repository.api.IRepositoryStructure;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public abstract class PublisherHandler {
```

Comment:
Create `IPublishHandler` interface to extract the abstract methods there:

```java
public abstract class PublisherHandler {
```

->

```java
public abstract class AbstractPublishHandler implements IPublishHandler {
```